src/features/product-manager/ProductList.jsx

```diff
@@ -1,15 +1,241 @@
+  return (
+    <CFormSelect onChange={(e) => setPageSelect(e.target.value)}>
+      {numberRow.map((item) => (
+        <option key={item} value={item} type="number">
+          {item}
+        </option>
+      ))}
+    </CFormSelect>
+  );
+}
+
+function FormInput({ control, name }) {
+  return (
+    <Controller
+      control={control}
+      name={name}
+      render={({ field: { onChange, onBlur, value, ref } }) => (
+        <CFormInput
+          onBlur={onBlur} // notify when input is touched
+          onChange={onChange} // send value to hook form
+          checked={value}
+          inputRef={ref}
+        />
+      )}
+    />
+  );
+}
+
+// eslint-disable-next-line no-unused-vars
+function FormProduct({ id, visible, setVisible }) {
+  const { handleSubmit, control } = useForm();
+  const onSubmit = (data) => {
+    // eslint-disable-next-line no-console
+    console.log(data);
+  };
+
+  return (
+    <CForm onSubmit={() => handleSubmit(onSubmit)}>
+      <CModal visible={visible} onClose={() => setVisible(false)}>
+        <CModalHeader onClose={() => setVisible(false)}>
+          <CModalTitle>{id}</CModalTitle>
+        </CModalHeader>
+
+        <CModalBody>
+          <CRow className="mb-3">
+            <CFormLabel className="col-sm-2 col-form-label">title</CFormLabel>
+            <CCol sm={10}>
+              <FormInput name="title" control={control} />
+            </CCol>
+          </CRow>
+        </CModalBody>
+        <CModalFooter>
+          <CButton color="secondary" onClick={() => setVisible(false)}>
+            Close
+          </CButton>
+          <CButton color="primary">Save changes</CButton>
+        </CModalFooter>
+      </CModal>
+    </CForm>
+  );
+}
 
 export default function ProductList() {
+  const numberRow = [3, 5, 10];
+  const headerTable = ['title', 'price', 'quantity', 'status'];
+
+  const [activePage, setActivePage] = useState(1);
+  const [pageSelect, setPageSelect] = useState(numberRow[0]);
+
   const dispatch = useDispatch();
-  const { product } = useSelector((state) => state.product);
+  const { products, loading, dispatchProduct, pagination } = useSelector((state) => state.product);
 
   useEffect(() => {
-    dispatch(getProducts());
-  }, []);
+    dispatch(getProducts({ page: 2, limit: 3 }));
+  }, [dispatchProduct]);
+
+  useEffect(() => {
+    dispatch(getProducts({ limit: pageSelect, page: activePage }));
+  }, [pageSelect]);
+
+  useEffect(() => {
+    dispatch(getProducts({ limit: pageSelect, page: activePage }));
+  }, [activePage]);
+
+  return (
+    <div>
+      <FilterBar />
+
+      {loading ? (
+        <ProcessLoading />
+      ) : (
+        <>
+          <Table headerTable={headerTable} data={products} />
+
+          <div className="d-flex justify-content-between">
+            <PginationComponent
+              activePage={activePage}
+              setActivePage={setActivePage}
+              pages={pagination.totalCount / pageSelect}
+              display={5}
```

Comment:
use maxDisplayNodePagination instead of display

Reply:
done